saged/utils.py

```diff
@@ -501,6 +542,32 @@ def calculate_loss_weights(dataset: "datasets.RefineBioLabeledDataset") -> torch
     return weights
 
 
+def calculate_skl_class_weights(dataset: "datasets.RefineBioLabeledDataset") -> Dict[int, float]:
```

Comment:
Just for future reference, I think you might be doing the same thing here as the `class_weight='balanced'` option that's built into sklearn. Probably would be good to check that it's the same if you decide to use that option, though (I may be totally wrong).

Reply:
That's right! The issue with `class_weight='balanced'` is that it doesn't work with the `partial_fit` function. As a result, I wrote this function to implement the same weighting scheme for models using `partial_fit` (I don't have any at the moment, but the first variants of `sag_skl` and `sgd_lr` did)